Week4/app/src/main/java/com/em4n0101/mymovies/MoviesAdapter.kt

```diff
@@ -0,0 +1,30 @@
+package com.em4n0101.mymovies
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.em4n0101.mymovies.data.Movie
+import com.em4n0101.mymovies.utils.MovieViewHolder
+
+class MoviesAdapter(private val movieList: ArrayList<Movie>, val delegate: SelectItemListener): RecyclerView.Adapter<MovieViewHolder>() {
+
+    interface SelectItemListener {
```

Comment:
Creating interface for click listeners is fine, but you can also do that by using lambdas. I will show you on the next session how to do this.